rmw_connext_cpp/src/functions.cpp

```diff
@@ -367,19 +367,46 @@ rmw_create_publisher(
     goto fail;
   }
 
+  datawriter_qos.reliability.max_blocking_time.sec = 2;
+  datawriter_qos.reliability.max_blocking_time.nanosec = 0;
+
+  switch (qos_policy.history) {
+    case RMW_QOS_POLICY_KEEP_LAST_HISTORY:
+      datawriter_qos.history.kind = DDS_KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_KEEP_ALL_HISTORY:
+      datawriter_qos.history.kind = DDS_KEEP_ALL_HISTORY_QOS;
+      break;
+    default:
+      RMW_SET_ERROR_MSG("Unknown QoS history policy");
+      goto fail;
+  }
+
+  switch (qos_policy.reliability) {
+    case RMW_QOS_POLICY_BEST_EFFORT:
+      datawriter_qos.reliability.kind = DDS_BEST_EFFORT_RELIABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_RELIABLE:
+      datawriter_qos.reliability.kind = DDS_RELIABLE_RELIABILITY_QOS;
+      break;
+    default:
+      RMW_SET_ERROR_MSG("Unknown QoS reliability policy");
+      goto fail;
+  }
+
   // ensure the history depth is at least the requested queue size
   assert(datawriter_qos.history.depth >= 0);
   if (
     datawriter_qos.history.kind == DDS::KEEP_LAST_HISTORY_QOS &&
-    static_cast<size_t>(datawriter_qos.history.depth) < queue_size
+    static_cast<size_t>(datawriter_qos.history.depth) < qos_policy.depth
   )
   {
-    if (queue_size > (std::numeric_limits<DDS_Long>::max)()) {
+    if (qos_policy.depth > (std::numeric_limits<DDS_Long>::max)()) {
       RMW_SET_ERROR_MSG(
         "failed to set history depth since the requested queue size exceeds the DDS type");
       goto fail;
     }
-    datawriter_qos.history.depth = static_cast<DDS_Long>(queue_size);
+    datawriter_qos.history.depth = static_cast<DDS::Long>(qos_policy.depth);
```

Comment:
Why switch the type here (as well as below)? The rest of the file uses `DDS_Long`.